src/backend/projects/project_routes.py

```diff
@@ -87,23 +97,64 @@ async def create_project(
 ):
     """Create a project in ODK Central and the local database"""
     odkproject = central_crud.create_odk_project(project_info.project_info.name)
-    # Use the ID we get from Central, as it's needed for many queries
-    project_info.project_info.id = odkproject['id']
     # TODO check token against user or use token instead of passing user
-    project = project_crud.create_project_with_project_info(db, project_info)
-    return project
+    project = project_crud.create_project_with_project_info(db, project_info, odkproject['id'])
+    # FIXME: This should only be done once when starting, instead of for each project
+    xlsforms = project_crud.read_xlsforms(db, config_env['XLSFORMS_LIBRARY'])
+
+    if project:
+        return project
+    else:
+        raise HTTPException(status_code=404, detail="Project not found")
 
 
-@router.post("/beta/{project_id}/upload_zip", response_model=project_schemas.ProjectOut)
-async def upload_beta_project(
+@router.post("/beta/{project_id}/upload", response_model=project_schemas.ProjectOut)
+async def upload_project_boundary_with_zip(
     project_id: int,
     project_name_prefix: str,
     task_type_prefix: str,
     upload: UploadFile,
     db: Session = Depends(database.get_db),
 ):
+    """This uploads the projecy boundary polygon to an existing project"""
     # TODO: consider replacing with this: https://stackoverflow.com/questions/73442335/how-to-upload-a-large-file-%e2%89%a53gb-to-fastapi-backend/73443824#73443824
-    project = project_crud.update_project_with_upload(
+    project = project_crud.update_project_with_zip(
```

Comment:
I assume this is going away soon?

Reply:
Not for a while on the backend, as it's what currently works. Now though with this PR there is a different process. which is automated. In the future it'll still be possible to upload the individual files, but the zip will be unneeded. But at this point there is no reason to depreciate working code.